shell/common/shell_test_platform_view_vulkan.cc

```diff
@@ -40,9 +39,128 @@ PointerDataDispatcherMaker ShellTestPlatformViewVulkan::GetDispatcherMaker() {
   };
 }
 
-// |GPUSurfaceVulkanDelegate|
-fml::RefPtr<vulkan::VulkanProcTable> ShellTestPlatformViewVulkan::vk() {
-  return proc_table_;
+ShellTestPlatformViewVulkan::OffScreenSurface::OffScreenSurface(
+    fml::RefPtr<vulkan::VulkanProcTable> vk)
+    : valid_(false), vk_(std::move(vk)) {
+  if (!vk_ || !vk_->HasAcquiredMandatoryProcAddresses()) {
+    FML_DLOG(ERROR) << "Proc table has not acquired mandatory proc addresses.";
+    return;
+  }
+
+  // Create the application instance.
+  std::vector<std::string> extensions = {};
+
+  application_ = std::make_unique<vulkan::VulkanApplication>(
+      *vk_, "FlutterTest", std::move(extensions));
+
+  if (!application_->IsValid() || !vk_->AreInstanceProcsSetup()) {
+    // Make certain the application instance was created and it setup the
+    // instance proc table entries.
+    FML_DLOG(ERROR) << "Instance proc addresses have not been setup.";
+    return;
+  }
+
+  // Create the device.
+
+  logical_device_ = application_->AcquireFirstCompatibleLogicalDevice();
+
+  if (logical_device_ == nullptr || !logical_device_->IsValid() ||
+      !vk_->AreDeviceProcsSetup()) {
+    // Make certain the device was created and it setup the device proc table
+    // entries.
+    FML_DLOG(ERROR) << "Device proc addresses have not been setup.";
+    return;
+  }
+
+  // Create the Skia GrContext.
+  if (!CreateSkiaGrContext()) {
+    FML_DLOG(ERROR) << "Could not create Skia context.";
+    return;
+  }
+
+  valid_ = true;
+}
+
+bool ShellTestPlatformViewVulkan::OffScreenSurface::CreateSkiaGrContext() {
+  GrVkBackendContext backend_context;
+
+  if (!CreateSkiaBackendContext(&backend_context)) {
+    FML_DLOG(ERROR) << "Could not create Skia backend context.";
+    return false;
+  }
+
+  sk_sp<GrContext> context = GrContext::MakeVulkan(backend_context);
+
+  if (context == nullptr) {
+    FML_DLOG(ERROR) << "Failed to create GrContext";
+    return false;
+  }
+
+  context->setResourceCacheLimits(vulkan::kGrCacheMaxCount,
+                                  vulkan::kGrCacheMaxByteSize);
```

Comment:
So this currently causes a test failure because on the GL backend we set the max byte size to 24MB, which doesn't match the Vulkan defaults. I think we want to keep using the Vulkan defaults but we will need to adjust the unit test appropriately. I'm good with landing this now and figuring out how to adjust the unit tests in the follow up PR to enable the unittests.